mlrun/datastore/targets.py

```diff
@@ -996,12 +999,22 @@ def get_dask_options(self):
     def as_df(self, columns=None, df_module=None, **kwargs):
         raise NotImplementedError()
 
+    def prepare_spark_df(self, df):
+        import pyspark.sql.functions as funcs
+
+        for col_name, col_type in df.dtypes:
+            if col_type.startswith("decimal("):
+                # V3IO does not support this level of precision
+                df = df.withColumn(col_name, funcs.col(col_name).cast("double"))
+        return df
+
     def write_dataframe(
         self, df, key_column=None, timestamp_key=None, chunk_id=0, **kwargs
     ):
         if hasattr(df, "rdd"):
```

Comment:
I guess "rdd" somehow means that it is spark engine but maybe we should add a comment so it would be more straight forward

Reply:
Yeah, that's what it means... It's done in a few other places in the code base and is not affected by this PR. Feel free to PR separately.